synapse/storage/databases/main/event_federation.py

```diff
@@ -1130,6 +1197,18 @@ def _delete_old_forward_extrem_cache_txn(txn):
             _delete_old_forward_extrem_cache_txn,
         )
 
+    async def insert_insertion_extremity(self, event_id: str, room_id: str) -> None:
+        await self.db_pool.simple_upsert(
+            table="insertion_event_extremities",
+            keyvalues={"event_id": event_id},
+            values={
+                "event_id": event_id,
+                "room_id": room_id,
+            },
+            insertion_values={},
+            desc="insert_insertion_extremity",
+        )
```

Comment:
Can we set `lock=false` on this? We're using `event_id` as the `keyvalues` which has a unique index constraint.

> `lock` should generally be set to True (the default), but can be set
>
> there is a UNIQUE INDEX on the key columns. In this case a conflict
          will cause an IntegrityError in which case this function will retry
          the update.

Reply:
Yes we can. Though the docstring is a bit misleading as we'll use native upserts (which ignores the `lock` param) for everything bar old SQLite versions, so it doesn't really matter in practice.

Reply:
Added and updated the docstring there to mention it only matters for SQLite 👍